### PR TITLE
fix(nuxt, nuxi): improve `pages` creation and removal dx

### PR DESCRIPTION
**Pull request by pi0**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Fixes some issues when creating and removing pages directory:

- When pages directory is empty (in all layers), and no explicit `options.pages` provided, module self-disables
- When pages is disabled, a placeholder `<NuxtPage>` with default slot takes in place and warns to create `pages/index.vue`
- Nuxi restarts when pages is disabled and there are some files inside pages OR when pages is enabled and there is no files in pages directory on watch events containing `pages/**` path

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/nuxi/src/commands/dev.ts [hunk 1]
```diff
@@ -129,16 +130,28 @@ export default defineNuxtCommand({
 
       const isDirChange = ['addDir', 'unlinkDir'].includes(event)
```
Comment (danielroe) [inline packages/nuxi/src/commands/dev.ts:131]: Probably not a problem with this PR, but I'm getting this when I remove the pages dir:

```
 ERROR  [unhandledRejection] ENOENT: no such file or directory, scandir '/playground/pages'

  at readdirSync (node:fs:1438:3)
  at FSWatcher.<anonymous> (packages/nuxi/src/commands/dev.ts:149:55)
  at FSWatcher.emit (node:events:513:28)
  at FSWatcher.emit (node:domain:489:12)
  at FSWatcher.emitWithAll (node_modules/.pnpm/chokidar@3.5.3/node_modules/chokidar/index.js:541:32)
  at FSWatcher._emit (node_modules/.pnpm/chokidar@3.5.3/node_modules/chokidar/index.js:632:8)
  at FSWatcher._remove (node_modules/.pnpm/chokidar@3.5.3/node_modules/chokidar/index.js:895:50)
  at FsEventsHandler.handleEvent (node_modules/.pnpm/chokidar@3.5.3/node_modules/chokidar/lib/fsevents-handler.js:267:16)
  at FsEventsHandler.checkExists (node_modules/.pnpm/chokidar@3.5.3/node_modules/chokidar/lib/fsevents-handler.js:255:12)
```

File: packages/nuxi/src/commands/dev.ts [hunk 2]
```diff
@@ -129,16 +130,28 @@ export default defineNuxtCommand({
...
+        }
+      }
+
+      if (file.startsWith(pagesDir)) {
+        const pagesDirFiles = readdirSync(pagesDir)
+        if (currentNuxt && !currentNuxt.options.pages && pagesDirFiles.length) {
+          return dLoad(true, 'Enabling pages...')
+        }
+        if (currentNuxt && currentNuxt.options.pages && !pagesDirFiles.length) {
+          return dLoad(true, 'Disabling pages...')
```
Comment (danielroe) [inline packages/nuxi/src/commands/dev.ts:154]: if you create a pages dir with no content, nothing happens. ✅

But if you then delete it you get the message 'Disabling pages' which is confusing as it was never enabled. It's a bit complex to decide whether there has been a change in whether the pages module was enabled or not.

Maybe also it is worth not displaying any extra logs in any case?

File: packages/nuxt/src/pages/module.ts [hunk 3]
```diff
@@ -18,16 +18,35 @@ export default defineNuxtModule({
...
     // Disable module (and use universal router) if pages dir do not exists or user has disabled it
-    if ((nuxt.options.pages === false || (nuxt.options.pages !== true && !pagesDirs.some(dir => existsSync(dir)))) && !isRouterOptionsPresent) {
+    const isNonEmptyDir = (dir: string) => existsSync(dir) && readdirSync(dir).length
+    const isPagesEnabled = () => {
+      if (nuxt.options.pages === false) {
+        return false
+      }
+      if (nuxt.options.pages === true) {
+        return true
+      }
```
Comment (danielroe) [inline packages/nuxt/src/pages/module.ts:29]:
```suggestion
      if (typeof nuxt.options.pages === 'boolean') {
        return nuxt.options.pages
      }
```

File: packages/nuxi/src/commands/dev.ts (hunk 2 above)
Comment (pi0) [reply packages/nuxi/src/commands/dev.ts:154]: I'm fine with hiding extra log but it won't happen with "disabling pages". Note that we override `pages` with actual state inside module.

**Comment by Atinux** [comment]
Beautiful work :)

Is there a chance we can get the logs from the Vue app on SSR and apply the same formatting as we do with consola?

File: packages/nuxt/src/pages/runtime/page-placeholder.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { defineComponent } from 'vue'
+
+export default defineComponent({
+  name: 'NuxtPage',
+  setup (_, props) {
+    if (process.dev) {
+      console.warn('Please create `pages/index.vue` to use `<NuxtPage>`')
```
Comment (Atinux) [inline packages/nuxt/src/pages/runtime/page-placeholder.ts:7]:
```suggestion
      console.warn('Create a Vue component in the `pages/` directory to enable `<NuxtPage>`')
```

**Review by Atinux — APPROVED** [review]

**Pull request merged**